src/corona_hakab_model/analyzers/state_machine_analysis.py

```diff
@@ -0,0 +1,169 @@
+                               population_size: int,
+                               state_counter: Counter) -> Tuple[List]:
+    """
+
+    :param medical_state_machine: State machine representing the medical state
+    :param population_size: The number of agents
+    :param state_counter: Counter that represents total number of days
+                          all the agents spent in the state overall
+    :return: Two dictionaries:
+             * The first one represents the empirical mean time spent in state,
+               conditioned that we visit it
+             * The second one represents the empirical mean time spent in state,
+               taking in consideration the total population
+    """
+    average_state_time_duration = dict()
+    state_duration_expected_time = dict()
+    for m in medical_state_machine.states:
+        if len(m.ever_visited) == 0:
+            average_state_time_duration[m.name] = 0
+            state_duration_expected_time[m.name] = 0
+        else:
+            average_state_time_duration[m.name] = state_counter[m.name] / len(m.ever_visited)
+            state_duration_expected_time[m.name] = state_counter[m.name] / population_size
+    return average_state_time_duration, state_duration_expected_time
+
+
+def _is_terminal_state(state):
+    """
+
+    :param state: A state of the medical state machine
+    :return: Whether this agent reached a state from which he will not infect anymore.
+             In our scenario it means he died or recovered.
+    """
+    return isinstance(state, TerminalState) and isinstance(state, ImmuneState)
+
+
+def monte_carlo_state_machine_analysis(configuration: Dict) -> Dict:
+    """
+
+    :param configuration: Dictionary with configuration for the mc run
+           configuration must contain:
+           * population_size for the mc
+           configuration might contain:
+           * consts_file - For loading Consts()
+           * circle_consts file - for loading CircleConsts
+    :return: Dictionary with statistics of the run:
+                * population_size
+                * days_passed - time it took to all the agents to recover/die
+                * time_in_each_state - for each state, total number of days all agents were in it
+                * visitors_in_each_state - Number of agents that visited each state
+                * average_duration_in_state - Empirical mean time to stay
+                                              at state conditioned that we enter it
+                * state_duration_expected_time - Empirical mean to stay in state, not conditioned
+                * average_time_to_terminal -Empirical mean time until death/recovery
+
+    """
+    if 'consts_file' in configuration:
+        consts = Consts.from_file(configuration['consts_file'])
+    else:
+        consts = Consts()
+
+    if "circle_consts_file" in configuration:
+        consts = CirclesConsts.from_file(configuration['circle_consts_file'])
```

Comment:
should be circle_const, no?

Reply:
you are right, I fixed this but forgot to push